src/resolver.js

```diff
@@ -36,11 +40,11 @@ export default function resolve({http, fetch, spec, url, baseDoc, mode, allowMet
   }
   else {
     // Store the spec into the url provided, to cache it
-    plugins.refs.docCache[baseDoc] = spec
+    docCache[baseDoc] = spec
```

Comment:
Yes, we should avoid global. The `docCache` here is created inside `resolve()` method, I suggest we move that inside `mapSpec()`. Basically just encapsulate the cache creation inside SpecMap, `resolve()` doesn't have to know about the cache at all. Therefore, there's no performance difference whatsoever.